File: 8kyu 3/summation.js
/*
|--------------------------------------------------------------------------
| Grasshopper - Summation
|--------------------------------------------------------------------------
|
*/

function summation(num) {
   let result = 0;
   for (i = 1; i <= num; i++) {
      hasil = result += i;
   }
   return hasil;
}


const summation = n => n * (n + 1) / 2;


console.log(summation(1), 1)
console.log(summation(8), 36)

File: Python/8py 3/summation.py
"""
|--------------------------------------------------------------------------
| Grasshopper - Summation
|--------------------------------------------------------------------------
|
"""

def summation(num):
   res = 0
   for i in range(1, num+1):
      res += i
   return res


summation=lambda n:n*-~n>>1


def summation(num):
   return num*(num+1) / 2


summation = lambda x: sum(range(1, x + 1))


def summation(num):
   return sum(range(1,num+1))
    
    
print(summation(1), 1)
print(summation(8), 36)
print(summation(22), 253)
print(summation(100), 5050)
print(summation(213), 22791)